pkg/controller/kogitomgmtconsole/kogitomgmtconsole_controller.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2020 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kogitomgmtconsole
+
+import (
+	kogitocli "github.com/kiegroup/kogito-cloud-operator/pkg/client"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/framework"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/infrastructure/services"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/logger"
+	imagev1 "github.com/openshift/api/image/v1"
+	routev1 "github.com/openshift/api/route/v1"
+	appsv1 "k8s.io/api/apps/v1"
+
+	appv1alpha1 "github.com/kiegroup/kogito-cloud-operator/pkg/apis/app/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+var log = logger.GetLogger("mgmtconsole_controller")
+
+var watchedObjects = []framework.WatchedObjects{
+	{
+		GroupVersion: routev1.GroupVersion,
+		AddToScheme:  routev1.Install,
+		Objects:      []runtime.Object{&routev1.Route{}},
+	},
+	{
+		GroupVersion: imagev1.GroupVersion,
+		AddToScheme:  imagev1.Install,
+		Objects:      []runtime.Object{&imagev1.ImageStream{}},
+	},
+	{
+		Objects: []runtime.Object{&corev1.Service{}, &appsv1.Deployment{}},
+	},
+	{
+		GroupVersion: routev1.GroupVersion,
+		AddToScheme:  routev1.Install,
+		Objects:      []runtime.Object{&routev1.Route{}},
+		Owner:        &appv1alpha1.KogitoDataIndex{},
+	},
+	{
+		Objects: []runtime.Object{&corev1.Service{}},
+		Owner:   &appv1alpha1.KogitoDataIndex{},
```

Comment:
Should the `KogitoDataIndex` instance also be added to the [OwnerReferences](https://github.com/kiegroup/kogito-cloud-operator/blob/17246b3b5be38b80ff55361d8e4b22cda6efae7d/pkg/infrastructure/services/deployer.go#L175) of the `Service` and `Route`, in order for this to work?

Reply:
Not needed. Once we have the `Service` and `Route` created in Data Index, it will be created with the correct `ownerReference`.

So the Management Console controller will be able to watch it.